src/app.js

```diff
@@ -81,8 +81,16 @@ export function app(app) {
             }).data
           )
 
-          if (result != null && result.then == null) {
-            repaint((state = merge(state, emit("update", result))))
+          function update(data) {
+            if (data != null) {
+              repaint((state = merge(state, emit("update", data))))
+            }
+          }
+
+          if (result != null && typeof result.then == "function") {
+            result.then(update)
+          } else {
+            update(result)
           }
```

Comment:
According to the slack discussion, we move from `result.then != null` to `typeof result.then == "function` because : 
> At least test if then is a function ?
> If the function return nothing, the `result != null` will silent ignore it. But if then is not a function, that will break the thing